### PR TITLE
Set selected context for bulk actions

### LINKED ISSUE #663: Bulk actions are missing "selected" context
Regression since we refactored the search results layout.

### PR DESCRIPTION
**Pull request by eyeseast**
Closes #663 

Related: the search API doesn't let us expand `projects`, so there's no way to pre-populate common projects without re-fetching. Going to open a new backend issue for that.

### CODE REVIEW

File: src/lib/components/forms/Projects.svelte [hunk 1]
```diff
@@ -33,11 +33,13 @@ and we don't want to do that everywhere.
...
+  $: console.log(documents);
+
   $: common = new Set(
     intersection(
-      documents.map((d) => d.projects),
+      documents.map((d) => d.projects ?? []),
       (a, b) => {
-        return a.id === b.id;
+        return a.id === b?.id;
       },
```
Comment (allanlasser) [inline src/lib/components/forms/Projects.svelte:43]: If we enable the `noUncheckedIndexedAccess` TS rule, these accesses should get flagged as type errors. Of course that will create a ton of work to fix, but should prevent bugs like this in the future.

**Review by allanlasser — APPROVED** [review]

File: src/lib/components/forms/Projects.svelte (hunk 1 above)
Comment (eyeseast) [reply src/lib/components/forms/Projects.svelte:43]: Might be worth doing. I wish the API were more consistent between Solr and Postgres.

Comment (allanlasser) [reply src/lib/components/forms/Projects.svelte:43]: I'll make an issue for it. I think there's a bunch of optimization and code quality tasks we can tackle post-launch.

Comment (allanlasser) [reply src/lib/components/forms/Projects.svelte:43]: #674 and #675 😎

**Pull request merged**